src/marqo/tensor_search/throttling/check_and_increment.lua

```diff
@@ -0,0 +1,26 @@
+local set_key = KEYS[1]
+
+local thread_name = ARGV[1]
+local thread_limit = tonumber(ARGV[2])
+local expire_time = tonumber(ARGV[3])
+

```

Comment:
Not blocking, but we should try get all times from within redis (no passed from Marqo). If, in the future, we decide that Redis should be remote, then we could get weird behaviour if Marqo server time is off from Redis time

Reply:
Can you create an issue/comment mentioning this?

Reply:
as for this, expire_time is the length of time we want a thread to be up, not the actual time!